python/ray/experimental/multiprocessing/pool.py

```diff
@@ -0,0 +1,493 @@
+        new_chunk_id = self._pool._submit_chunk(self._func, self._iterator,
+                                                self._chunksize, actor_index)
+        self._submitted_chunks.append(False)
+        self._result_thread.add_object_id(new_chunk_id)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        # Should be implemented by subclasses.
+        raise NotImplementedError
+
+
+class OrderedIMapIterator(IMapIterator):
+    def next(self, timeout=None):
+        if len(self._ready_objects) == 0:
+            if self._next_chunk_index == self._total_chunks:
+                raise StopIteration
+
+            while timeout is None or timeout > 0:
+                start = time.time()
+                index = self._result_thread.next_ready_index(timeout=timeout)
+                self._submit_next_chunk()
+                self._submitted_chunks[index] = True
+                if index == self._next_chunk_index:
+                    break
+                if timeout is not None:
+                    timeout = max(0, timeout - (time.time() - start))
+
+            while self._next_chunk_index < len(
+                    self._submitted_chunks
+            ) and self._submitted_chunks[self._next_chunk_index]:
+                for result in self._result_thread.result(
+                        self._next_chunk_index):
+                    self._ready_objects.append(result)
+                self._next_chunk_index += 1
+
+        return self._ready_objects.popleft()
+
+
+class UnorderedIMapIterator(IMapIterator):
+    def next(self, timeout=None):
+        if len(self._ready_objects) == 0:
+            if self._next_chunk_index == self._total_chunks:
+                raise StopIteration
+
+            index = self._result_thread.next_ready_index(timeout=timeout)
+            self._submit_next_chunk()
+
+            for result in self._result_thread.result(index):
+                self._ready_objects.append(result)
+            self._next_chunk_index += 1
+
+        return self._ready_objects.popleft()
+
+
+@ray.remote
+class PoolActor(object):
+    def __init__(self, initializer=None, initargs=None):
+        if initializer:
+            initargs = initargs or ()
+            initializer(*initargs)
+
+    def ping(self):
+        # Used to wait for this actor to be initialized.
+        pass
+
+    def run_batch(self, func, batch):
+        results = []
+        for args, kwargs in batch:
+            args = args or ()
+            kwargs = kwargs or {}
+            try:
+                results.append(func(*args, **kwargs))
+            except Exception as e:
+                results.append(PoolTaskError(e))
+        return results
+
+
+# https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing.pool
+# TODO(edoakes): cloudpickle can't pickle generator objects.
+class Pool(object):
```

Comment:
Can you have something that shows up as a docstring? It'd be nice for users using an IDE not need to cross-reference to a different source.